dcore/advection.py

```diff
@@ -185,7 +190,7 @@ def apply(self, x_in, x_out):
 
 class EmbeddedDGAdvection(Advection):
 
-    def __init__(self, state, V, Vdg, continuity):
+    def __init__(self, state, Vdg, continuity):
 
         super(EmbeddedDGAdvection, self).__init__(state)
         self.dgadvection = DGAdvection(state, Vdg, continuity)
```

Comment:
That is what I'd originally intended... colinjcotter, can you remember why you wanted it done this way?

________________________________
From: Lawrence Mitchell <notifications@github.com>
Sent: 04 May 2016 14:18:56
To: firedrakeproject/dcore
Cc: Shipton, Jemma; Author
Subject: Re: [firedrakeproject/dcore] Advection bugfixes (#33)


In dcore/advection.py<https://github.com/firedrakeproject/dcore/pull/33#discussion_r62036474>:

> @@ -185,7 +190,7 @@ def apply(self, x_in, x_out):
>
>  class EmbeddedDGAdvection(Advection):
>
> -    def __init__(self, state, V, Vdg, continuity):
> +    def __init__(self, state, Vdg, continuity):
>
>          super(EmbeddedDGAdvection, self).__init__(state)
>          self.dgadvection = DGAdvection(state, Vdg, continuity)


While you're here, I think you should inherit from DGAdvection and then you can simplify things:

class EmbeddedDGAdvection(DGAdvection):
    def __init__(self, state, Vdg, continuity):
        super(EmbeddedDGAdvection, self).__init__(state, Vdg, continuity)
        self.xdg_in = Function(Vdg)
        self.xdg_out = Function(Vdg)

    def apply(self, x_in, x_out):
        self.xdg_in.interpolate(x_in)
        # Call normal DG advection on DG fields via superclass
        super(EmbeddedDGAdvection, self).apply(self.xdg_in, self.xdg_out)
        x_out.project(self.xdg_out)

That way you're not carrying around an extra ubar that you need to move from this advection object to the dgadvection object inside it.

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/firedrakeproject/dcore/pull/33/files/03666bbe272832bcf3396124972c9bb0674f97af#r62036474>